Add gaussian filter for void fraction (#2012)
chaos-polymtl/lethe: doc/source/parameters/unresolved-cfd-dem/void-fraction.rst

* The ``l2 smoothing length`` is a smoothing length used for smoothing the L2 projection of the void fraction to avoid sharp discontinuities which can lead to instabilities in the simulation.
* The ``qcm smoothing length`` allows us to fix the characteristic length of the QCM filter in the simulation to a given value. When the ``spherical`` filter is used, half of this value is the radius of the averaging sphere; when the ``gaussian`` filter is used, half of this value is the standard deviation :math:`\sigma`. If this option is used (a value other than 0 is specified), it overrides the default calculation based on the local cell measure. The previous parameter name ``qcm sphere diameter`` is kept as a deprecated alias and remains accepted in ``.prm`` files.
* The ``qcm sphere equal cell volume`` determines whether or not we want to use a reference sphere with the same volume as the element in which it is located. If it is disabled, then each sphere will have a radius equal to the size of the element in which it is located. This parameter is important only when the ``qcm smoothing length`` is not used or is set to 0.

> * The ``qcm filter type`` selects the kernel used to weigh particle contributions onto the mesh in the ``qcm`` method. Two options are available: ``spherical`` (default) and ``gaussian``. The user always sets ``qcm smoothing length`` to the **diameter** of the averaging sphere (spherical filter) or to **2σ** (Gaussian filter); the code halves this value internally to obtain the kernel characteristic length. For each particle of radius :math:`r_p` and volume :math:`V_p`, the kernel value at distance :math:`d` from the particle centre to a quadrature point is evaluated and then normalised by the sum of all kernel values over the neighbour-cell stencil so that the total particle volume is conserved.

Reviewer: ```suggestion
* The ``qcm filter type`` selects the kernel used to weigh particle contributions onto the mesh in the ``qcm`` method. Two options are available: ``spherical`` (default) and ``gaussian``. The user always sets ``qcm smoothing length`` to the **diameter** of the averaging sphere (spherical filter) or to **2σ** (Gaussian filter); the code halves this value internally to obtain the kernel characteristic length (i.e., the radius or the standard deviation respectively). For each particle of radius :math:`r_p` and volume :math:`V_p`, the kernel value at distance :math:`d` from the particle centre to a quadrature point is evaluated and then normalised by the sum of all kernel values over the neighbour-cell stencil so that the total particle volume is conserved.
```

>   * **Spherical** (``spherical``): the kernel is the intersection volume between the particle sphere (radius :math:`r_p`) and an averaging sphere of radius :math:`\ell = \tfrac{1}{2}\,d_{\text{filter}}` centred at the quadrature point. In 3-D, for an overlap distance :math:`0 < d < r_p + \ell`, this is

Reviewer: ```suggestion
  * **Spherical** (``spherical``): the kernel is the intersection volume between the particle sphere (radius :math:`r_p`) and an averaging sphere of radius :math:`r_{\mathrm{avg}} = \tfrac{1}{2}\,d_{\text{filter}}` centred at the quadrature point. In 3-D, for an overlap distance :math:`0 < d < r_p +r_{\mathrm{avg}}`, this is
```

    .. math::

>        K_{\text{sph}}(r_p,\ell,d) = \frac{\pi\,(r_p + \ell - d)^2

Reviewer: ```suggestion
       K_{\text{sph}}(r_p,r_{\mathrm{avg}},d) = \frac{\pi\,(r_p + r_{\mathrm{avg}}- d)^2
```

>          \left(d^2 + 2d\,r_p - 3r_p^2 + 2d\,\ell + 6\,r_p\,\ell - 3\ell^2\right)}

Reviewer: ```suggestion
         \left(d^2 + 2d\,r_p - 3r_p^2 + 2d\,r_{\mathrm{avg}} + 6\,r_p\,r_{\mathrm{avg}} - 3r_{\mathrm{avg}}^2\right)}
```

         {12\,d}

>     and zero when :math:`d \geq r_p + \ell`, or :math:`V_p` when :math:`d \leq \ell - r_p`.

Reviewer: ```suggestion
    and zero when :math:`d \geq r_p +r_{\mathrm{avg}}`, or :math:`V_p` when :math:`d \leq r_{\mathrm{avg}} - r_p`.
```
